geometry_msgs/Point agent_position
geometry_msgs/Vector3 agent_velocity
float64 agent_size

# dynamic obstacles
geometry_msgs/Vector3[] obs_position
geometry_msgs/Vector3[] obs_velocity
geometry_msgs/Vector3[] obs_size

# static obstacles
float64[] laser_points
float64 laser_range
float64 laser_res # use the larger one among horizonal and vertical

float64 max_velocity
geometry_msgs/Vector3 rl_velocity
---
geometry_msgs/Vector3 safe_action